storage_service/locations/tests/test_gpg.py

```diff
@@ -0,0 +1,498 @@
+                  expected='fail'),
+        CrTarCase(path='/a/b/c', isfile=False, istar=True, raises=False,
+                  expected='fail'),
+        CrTarCase(path='/a/b/c', isfile=False, istar=False, raises=True,
+                  expected='fail')
+    ]
+)
+def test__create_tar(mocker, path, will_be_file, will_be_tar, sp_raises,
+                     expected):
+    if sp_raises:
+        mocker.patch.object(subprocess, 'check_output',
+                            side_effect=OSError('gotcha!'))
+    else:
+        mocker.patch.object(subprocess, 'check_output')
+    mocker.patch.object(os.path, 'isfile', return_value=will_be_file)
+    mocker.patch.object(tarfile, 'is_tarfile', return_value=will_be_tar)
+    mocker.patch.object(os, 'rename')
+    mocker.patch.object(shutil, 'rmtree')
+    fixed_path = path.rstrip('/')
+    tarpath = '{}.tar'.format(fixed_path)
+    if expected == 'success':
+        ret = gpg._create_tar(path)
+        shutil.rmtree.assert_called_once_with(fixed_path)
+        os.rename.assert_called_once_with(tarpath, fixed_path)
+        tarfile.is_tarfile.assert_any_call(fixed_path)
+        assert ret is None
+    else:
+        with pytest.raises(gpg.GPGException) as excinfo:
+            ret = gpg._create_tar(path)
+        assert 'Failed to create a tarfile at {} for dir at {}'.format(
+            tarpath, fixed_path) == str(excinfo.value)
+        assert not shutil.rmtree.called
+        assert not os.rename.called
+    if not sp_raises:
+        os.path.isfile.assert_called_once_with(tarpath)
+        if will_be_file:
+            tarfile.is_tarfile.assert_any_call(tarpath)
+
+
+class TestGPG(TestCase):
+
+    fixtures = ['base.json', 'package.json', 'gpg.json']
+
+    def test__encr_path2key_fingerprint(self):
+        package = Package.objects.get(pk=8)
+        exp_curr_path = (
+            'some/relative/path/to/'
+            'images-transfer-abcdabcd-97dd-48e0-8417-03be78359531')
+        assert package.current_path == exp_curr_path
+        assert package.encryption_key_fingerprint == EXP_FINGERPRINT
+
+        encr_path = exp_curr_path
+        assert gpg._encr_path2key_fingerprint(encr_path) == EXP_FINGERPRINT
+
+        encr_path = '/abs/path/to/{}'.format(exp_curr_path)
+        assert gpg._encr_path2key_fingerprint(encr_path) == EXP_FINGERPRINT
+
+        encr_path = '{}/data/objects/somefile.jpg'.format(exp_curr_path)
+        assert gpg._encr_path2key_fingerprint(encr_path) == EXP_FINGERPRINT
+
+        encr_path = '/abs/path/to/{}/data/objects/somefile.jpg'.format(
+            exp_curr_path)
+        assert gpg._encr_path2key_fingerprint(encr_path) == EXP_FINGERPRINT
+
+        with pytest.raises(gpg.GPGException) as excinfo:
+            encr_path = '/some/non/matching/path.jpg'
+            gpg._encr_path2key_fingerprint(encr_path)
+        assert 'Unable to find package matching encrypted path {}'.format(
+            encr_path) in str(excinfo.value)
```

Comment:
Impressive tests!

One minor detail, some of them are declared with two underscores and other with only one, is there a reason for that? If not, I think we usually use only one.

Reply:
The two underscores is actually a consistency because I'm naming each test func with `test_` followed by the name of the function/method being tested. The two underscores cases arise because the function/method being tested has a name that starts with an underscore.

Reply:
Great, thanks!